contributed_definitions/nyaml/NXxps.yaml

```diff
@@ -0,0 +1,451 @@
+category: application
+doc: |
+  This is the application definition for X-ray photoelectron spectroscopy.
+type: group
+NXxps(NXmpes):
+  (NXentry):
+    definition(NX_CHAR):
+      enumeration: [NXxps]
+    method:
+      exists: recommended
+      doc: |
+        A name of the experimental method according to `Clause 11`_ of
+        the `ISO 18115-1:2023`_ specification.
+        
+        Examples for XPS-related experiments include:
+            * X-ray photoelectron spectroscopy (XPS)
+            * angle-resolved X-ray photoelectron spectroscopy (ARXPS)
+            * ultraviolet photoelectron spectroscopy (UPS)
+            * hard X-ray photoemission spectroscopy (HAXPES)
+            * near ambient pressure X-ray photoelectron spectroscopy (NAPXPS)
+            * electron spectroscopy for chemical analysis (ESCA)
+        
+        .. _ISO 18115-1:2023: https://www.iso.org/standard/74811.html
+        .. _Clause 11: https://www.iso.org/obp/ui/en/#iso:std:iso:18115:-1:ed-3:v1:en:sec:11
+    xps_coordinate_system(NXcoordinate_system):
+      doc: |
+        Description of the XPS base coordinate system, which is defined such that the positive z-axis
+        points along the sample normal, and the x- and y-axes lie within the sample surface.
+        
+        The fields `x`, `y`, `z`, and `scaling` are used to describe the orientation of the XPS
+        coordinate system with respect to the beam coordinate system (.).
+      origin:
+        enumeration: [sample surface]
+      handedness:
+        enumeration: [right_handed]
+      z_direction:
+        enumeration: [sample normal]
+      x(NX_NUMBER):
+      y(NX_NUMBER):
+      z(NX_NUMBER):
+      offset(NX_NUMBER):
```

Comment:
Put into NXcoordinate_system_set
Also, why restrict this so strongly? I think e.g. liquid jet people use different geometries.
That also means you coordinate system (expressed in MCstas) changes if you use different emission angles.
What does offset signify here?

Reply:
Ok, this was mostly me playing around with NXcoordinate_system after the discussion with Sandor yesterday.

My original idea was to use the coordinate system "traditionally" used in XPS, i.e., the origin is at the sample surface. The advantage would be that it enforces and `xps_coordinate_system` in the standard. You can always define another coordinate system if you are working differently (like in liquid jet). But then you are by definition not working in the default XPS CS.

However, as you correctly point out later, the angles that we are acutally interested in are not really describable by this CS, but rather MCstas works fine here. So there are two options:
1) keep this default CS, but reference the transformation anyway to MCstas
2) just drop this CS.